nav2_bt_navigator/behavior_trees/navigate_w_replanning.xml

```diff
@@ -8,7 +8,7 @@
       <RateController hz="1.0">
         <Fallback>
           <GoalReached/>
-          <ComputePathToPose goal="{goal}"/>
+          <ComputePathToPose goal="{goal}" planner_name="ComputePathToPose"/>
```

Comment:
I'm not blocking if you really think this is the correct way to go. However, I see the duplication of `ComputePathToPose` as an indication that something is not quite right.

I understand your intention that this `planner_name` field will map to some plugin (`planner_type`) that will actually compute that path. But I think we haven't clarified what is the meaning of the *key* in that mapping.

I see the following options:

1. Action name is `ComputePathToPose`, but now planner_name="nav2_navfn_planner/NavfnPlanner" and remove the mapping altoghether from the server.
2. Action name is `ComputePathToPose`, keep mapping in the server and clarify what is the meaning of the `planner_name` *key*. This action is always computing a path to a pose, so does this field refers to some property of the resulting path? e.g. "ShortestPath", "LongestPath", etc. Is so, then `planner_name` is not the correct name.
3. Change the action name to `ComputePath`, now the *key* should provide info if we're computing to a pose, covering some area, going to and back, etc.



`<ComputePathToPose goal=>

Reply:
I'd think the planner name should be the algorithm name which implements the` ComputePathToPose` action, so here it would be `planner_name="Navfn"`.

Reply:
3 was the original thing we discussed in the meeting and was shot down- so I wouldn’t like to revert that unless some critical mass of changes in people’s opinions. 

On the name, agreed, any other suggestion, maybe ComputeGridPath since its a grid based wavefront planner?

Reply:
Optionally, I can just add no xml key. No key & 1 plugin in server will default to the given plugin. Though this feature loses all self-documentation

Reply:
Here's option 2.1:

```xml
<ComputePathToPose goal="{goal}" path_property="GridBased"/>
```
Path property could mean anything lenght/optimality, what representation is used to compute, computation time, etc. but we require the property to plugin mapping to be specified on the `nav2_params.yaml`:

```yaml
planner_server:
  ros__parameters:
    path_properties: ["ShortestPath", "FastestComputation", "GridBased", "Safest"]
    path_property_plugin: ["some_package/MPC", "nav2_navfn_planner/NavfnPlanner", ...]
```

Reply:
I could live with `path_property`. Do you think `GridBased` is a good name to replace `ComputePathToPose`? Are these 2 names exclusive decisions or do you want them both paired.

Reply:
I guess `GridBased` is good enough for now.

> Are these 2 names exclusive decisions or do you want them both paired.

I'm not sure what you mean here. Could you elaborate?